proto/streamlit/proto/GraphVizChart.proto

```diff
@@ -26,5 +26,8 @@ message GraphVizChart {
   // A unique ID of this element.
   string element_id = 5;
 
+  // The engine used to layout and render the graph.
+  string engine = 6;
```

Comment:
nit: One question: is the engine information removed from whatever is in the `spec`? If it is still in the `spec` part, we might not even need to add a new proto property.

Reply:
Can you also change the comment above for spec, which currently says:

> A specification of the GraphViz graph in the "Dot" language.

since the `in the "Dot" language.` won't be correct anymore.

Reply:
> nit: One question: is the engine information removed from whatever is in the `spec`?

I thought about that.. `element.spec` is a string containing the DOT representation of the graph. The engine itself (e.g., "dot", "neato", "twopi", etc) is not typically part of the DOT language. Instead, the DOT language describes the structure and styling of the graph, and the engine determines how that graph is laid out. 

So unless we embed the engine information within the DOT string in some custom manner (which is non-standard), I'm not if we can infer the engine from `spec`. And if we remove the engine from the proto message, we'll have to find some other way to communicate this information to the frontend.

Reply:
> since the in the "Dot" language. won't be correct anymore.

The graphs are still specified in the dot language though, right? There's a difference between the [dot language](https://www.graphviz.org/doc/info/lang.html) which is the abstract grammar and the [dot layout engine](https://www.graphviz.org/docs/layouts/dot/) which is an algorithm 🤔

Reply:
> The graphs are still specified in the dot language though, right?

oh, I see. Didn't know that. Then you can forget that comment

Reply:
> The engine itself (e.g., "dot", "neato", "twopi", etc) is not typically part of the DOT language.

In that case, adding it as extra proto property as in the implementation is the best solution IMO :)